### PR TITLE
Feature/profile page

### PR DESCRIPTION
**Pull request by andela-kndungu**
#### What does this PR do?

Add a profile page to show a summary of a user's projects and time spent on each.

#### Description of Task to be completed?

Add a profile page with the following details:
1. User's name, email and join date.
2. Organisations user belongs to.
3. User's projects in each organisation.
4. Visualisation of the time spent on each project

#### How should this be manually tested?

After logging in click on the "Profile" item in the menu. The profile page should appear with your details.

#### What are the relevant pivotal tracker stories?

[Feature #118457197] (https://www.pivotaltracker.com/projects/1454736/stories/118457197)
screen shot 2016-06-27 at 12 40 34 pm

@andela-ladenusi

### CODE REVIEW

File: app/scripts/components/ProfilePage/details.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      user: React.PropTypes.shape({
+        name: React.PropTypes.string,
+        email: React.PropTypes.string,
+        created_at: React.PropTypes.string
+      })
+    },
+
+    render: function() {
+      var joinDate = new Date(this.props.user.created_at);
+      joinDate = joinDate.getDate() + '/' + joinDate.getMonth() + '/' + joinDate.getFullYear();
```
Comment (andela-kndungu) [inline app/scripts/components/ProfilePage/details.jsx]: Using Moment.js would have been easier

File: app/scripts/components/ProfilePage/Index.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    UserActions = require('../../actions/UserActions'),
+    ProfileStore = require('../../stores/ProfileStore'),
+
+    AuthoriseUserAccess = require('../../mixins/AuthoriseUserAccess');
+
+  var Profile = React.createClass({ // eslint-disable-line
+    mixins: [AuthoriseUserAccess],
+
+    getInitialState: function() {
+      return ({user: {}, orgs:{}});
```
Comment (andela-kndungu) [inline app/scripts/components/ProfilePage/Index.jsx]: orgs should be initialised as an empty array.

File: app/scripts/components/ProfilePage/Orgs.jsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  var React = require('react');
+
+  var Orgs = React.createClass({ // eslint-disable-line
+    propTypes: {
+      orgs: React.PropTypes.arrayOf(React.PropTypes.object)
+    },
+
+    render: function() {
+
+      var orgs = this.props.orgs[0],
```
Comment (andela-kndungu) [inline app/scripts/components/ProfilePage/Orgs.jsx]: Use .length property instead.

File: app/scripts/components/ProfilePage/Visualisations.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    PieChart = require('./PieChart.jsx');
+
+  var Visualisations = React.createClass({ // eslint-disable-line
+    propTypes: {
+      orgs: React.PropTypes.arrayOf(React.PropTypes.object)
+    },
+
+    render: function() {
+      var projects = [];
+      if (this.props.orgs[0]) {
```
Comment (andela-kndungu) [inline app/scripts/components/ProfilePage/Visualisations.jsx]: Use .length property instead

File: app/scripts/components/ProfilePage/Visualisations.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+          return org.projects;
+        });
+      }
+      projects = [].concat.apply([], projects);
+
+      var trimeNodes = projects.map(function(project) {
+        return (<PieChart key={project.id} projectId={project.id} projectName={project.name}/>);
+      });
+
+      if (trimeNodes[0])  {
```
Comment (andela-kndungu) [inline app/scripts/components/ProfilePage/Visualisations.jsx]: Use .length property instead.

File: app/scripts/components/ProfilePage/Visualisations.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      var projects = [];
+      if (this.props.orgs[0]) {
+        projects = this.props.orgs.map(function(org) {
+          return org.projects;
+        });
+      }
+      projects = [].concat.apply([], projects);
+
+      var trimeNodes = projects.map(function(project) {
+        return (<PieChart key={project.id} projectId={project.id} projectName={project.name}/>);
```
Comment (andela-kndungu) [inline app/scripts/components/ProfilePage/Visualisations.jsx]: You can use the value passed to key for the project ID

File: app/scripts/components/ProfilePage/PieChart.jsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
+(function() {
+  'use strict'; // eslint-disable-line
+  var React = require('react'),
+    PieChart = require('rd3').PieChart,
+    ChartStore = require('../../stores/ChartStore'),
+    ProjectActions = require('../../actions/ProjectActions');
+
+  var Chart = React.createClass({ // eslint-disable-line
+    propTypes: {
+      projectId: React.PropTypes.number,
```
Comment (andela-kndungu) [inline app/scripts/components/ProfilePage/PieChart.jsx]: Replace with `key`

File: app/scripts/components/ProfilePage/PieChart.jsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        timeTracks.forEach(function(track) {
+          start += track.startedAt/1000;
+          finish += track.finishedAt/1000;
+        });
+        return finish - start;
+      };
+
+      var totalTime= 0;
+
+      if(this.state.trimes[0]) {
```
Comment (andela-kndungu) [inline app/scripts/components/ProfilePage/PieChart.jsx]: Use length property instead

File: server/controllers/users.js [hunk 9]
```diff
@@ -169,6 +174,87 @@
...
+          return res.send(users);
+        })
+        .catch(function(err) {
+          return res.status(500).json({
+            error: err.message
+          });
+        });
+    },
+
+    getVisualData: function(req, res) {
```
Comment (andela-kndungu) [inline server/controllers/users.js]: Please delete

File: server/routes/users.js [hunk 10]
```diff
@@ -10,5 +10,8 @@
     app.post('/api/users/login', Users.login);
     app.get('/api/users/session', Users.authenticate, Users.session);
     app.put('/api/users/logout', Users.authenticate, Users.logout);
+    app.get('/api/users/:userId/orgs', Users.authenticate, Users.orgUsers);
+    app.get('/api/users/visual-info', Users.authenticate, Users.getVisualData);
```
Comment (andela-kndungu) [inline server/routes/users.js]: Please delete

File: app/scripts/components/ProfilePage/details.jsx (hunk 1 above)
Comment (andela-jmuturi) [inline app/scripts/components/ProfilePage/details.jsx]: Or perhaps `joinDate.toLocaleDateString()`?

**Pull request merged**